### PR TITLE
Change: "Dungeon Filter" toggle changed to list

### PR DESCRIPTION
**Pull request by byWambo**
Changed "Dungeon Filter" toggle to list, sorted the unsorted messages/patterns and added ambience as type.

grafik

Changelog: Dungeon Filter is now fully customizable -Wambo

I hope the colors are good 😄

### CODE REVIEW

**Comment by hannibal002** [comment]
The colors look good 👍

**Comment by byWambo** [comment]
Here we go:
grafik

File: src/main/java/at/hannibal2/skyhanni/config/features/chat/ChatConfig.java [hunk 1]
```diff
@@ -32,10 +36,34 @@ public class ChatConfig {
     public ChatSymbols chatSymbols = new ChatSymbols();
 
     @Expose
-    @ConfigOption(name = "Dungeon Filter", desc = "Hides pickup, reminder, buff, damage, ability, puzzle and end messages in Dungeons.")
-    @ConfigEditorBoolean
-    @FeatureToggle
-    public boolean dungeonMessages = true;
+    @ConfigOption(name = "Dungeon Filter", desc = "Hide specific message types in Dungeons.")
+    @ConfigEditorDraggableList()
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/config/features/chat/ChatConfig.java]: You can remove the () at the end

File: src/main/java/at/hannibal2/skyhanni/config/features/chat/ChatConfig.java [hunk 2]
```diff
@@ -32,10 +36,34 @@ public class ChatConfig {
...
+        START("§aClass Buffs §r/ §cMort Dialog"),
+        AMBIENCE("§bAmbience"),
+        PICKUP("§ePickup"),
+        REMINDER("§cReminder"),
+        BUFF("§dBlessings"),
+        NOT_POSSIBLE("§cNot possible"),
+        DAMAGE("§cDamage"),
+        ABILITY("§dAbilities"),
+        PUZZLE("§dPuzzle §r/ §cQuiz"),
+        END("§cEnd §a(End of run spam)");
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/config/features/chat/ChatConfig.java]: please use a trailing comma instead (for cleaner changelogs in the future)

File: src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonChatFilter.kt [hunk 3]
```diff
@@ -161,37 +160,36 @@ class DungeonChatFilter {
...
-        "buff" to buffPatterns,
-        "damage" to damagePatterns,
-        "ability" to abilityPatterns,
-        "puzzle" to puzzlePatterns,
-        "end" to endPatterns,
-    )
-    private val messagesEndsWithMap: Map<String, List<String>> = mapOf(
-        "end" to endMessagesEndWith,
+    private val messagesMap: Map<ChatConfig.DungeonMessageTypes, List<String>> = mapOf(
+        ChatConfig.DungeonMessageTypes.PREPARE to prepareMessages,
```
Comment (hannibal002) [inline src/main/java/at/hannibal2/skyhanni/features/dungeon/DungeonChatFilter.kt]: You can use typealias here to make all lines way shorter

**Review by hannibal002 — CHANGES_REQUESTED** [review]
If anything is unclear, please msg me on discord

File: src/main/java/at/hannibal2/skyhanni/config/features/chat/ChatConfig.java (hunk 1 above)
Comment (byWambo) [reply src/main/java/at/hannibal2/skyhanni/config/features/chat/ChatConfig.java]: removed in [Commit](https://github.com/hannibal002/SkyHanni/pull/884/commits/09dd5658c28d09b221a5c88ccdc4e12a6fc06111) ty

**Review by hannibal002 — APPROVED** [review]

**Pull request merged**